internal/queue/consumer.go

```diff
@@ -1,46 +1,76 @@
 package queue
 
-import "errors"
+import (
+	"errors"
+	"sublinks/sublinks-federation/internal/worker"
+)
 
-func (q *RabbitQueue) createConsumer(queueName string) error {
+func (q *RabbitQueue) createConsumer(queueData ConsumerQueue) error {
 	channelRabbitMQ, err := q.Connection.Channel()
 	if err != nil {
 		return err
 	}
-	err = q.createQueue(channelRabbitMQ, queueName)
+	err = q.createQueue(channelRabbitMQ, queueData.QueueName)
 	if err != nil {
 		return err
 	}
+	err = channelRabbitMQ.QueueBind(
+		queueData.QueueName,  // queue name
+		queueData.RoutingKey, // routing key
+		queueData.Exchange,   // exchange
+		false,
+		nil)
+	if err != nil {
+		return err
+	}
+
 	// Subscribing to QueueService1 for getting messages.
 	messages, err := channelRabbitMQ.Consume(
-		queueName, // queue name
-		"",        // consumer
-		true,      // auto-ack
-		false,     // exclusive
-		false,     // no local
-		false,     // no wait
-		nil,       // arguments
+		queueData.QueueName, // queue name
+		"",                  // consumer
+		false,               // auto-ack
```

Comment:
The manual ack logic I have right now has issues. If you Nack a message, it just shoves it back in the queue and you reattempt to process it next time it's turn comes up.  Need to investigate this more as it just creates an endless loop in local dev. I suspect there is a counter on the message somewhere that you can use to see how many times you've seen it. Also, a "retry after X period of time" would be nice and be able to apply an exponential backoff strategy

Reply:
We should really implement a dead letter exchange. Let me know what you think? It would be beneficial for issues like this.